### PR TITLE
Add basic metrics to theia

### PR DESCRIPTION
**Pull request by hexa00**
This patch adds basic metrics to theia using prometheus.
See: https://github.com/siimon/prom-client for more information.

Fixes: #680

Signed-off-by: Antoine Tremblay <antoine.tremblay@ericsson.com>

### CODE REVIEW

File: packages/metrics/compile.tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "extends": "../../configs/base.tsconfig",
+  "compilerOptions": {
+    "rootDir": "src",
+    "outDir": "lib",
+    "baseUrl": ".",
+    "paths": {
+      "humane-js": [
```
Comment (akosyakov) [inline packages/metrics/compile.tsconfig.json]: it is something @theia/core specific

Comment (hexa00) [reply packages/metrics/compile.tsconfig.json]: oops thx fixing

**Comment by svenefftinge** [comment]
How are metrics visualized, i.e. who uses that exposed rest service?

**Comment by hexa00** [comment]
This is used by https://prometheus.io/

For graph and alerts etc too

**Pull request merged**